File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
  int area = brown + yellow;
  vector<int> answer;

  for (int height, width, i = 1; i * i <= area; i++) {
    if (area % i) continue;
    height = i;
    width = area / i;
    if (2 * (height + width) - 4 == brown) {
      answer.push_back(width);
      answer.push_back(height);
      break;
    }
  }
  return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {

    public int[] solution(int brown, int yellow) {
        int area = brown + yellow;
        int[] answer = {0, 0};

        for (int height = 1; height * height <= area; height++) {
            if (area % height == 0) {
                int width = area / height;

                if (2 * (width + height) - 4 == brown) {
                    answer[0] = width;
                    answer[1] = height;
                    break;
                }
            }
        }
        return answer;
    }
}